docs/data/advanced-components/overview.md

```diff
@@ -0,0 +1,172 @@
+
+We are using a commercial license to forward the development of the most advanced features, where the MIT model can't sustain it.
+A solution to a problem should only be commercial if it has no MIT alternatives.
+
+For each set of component, you can check the feature comparison table:
+
+- [Data Grid](/x/react-data-grid/getting-started/#feature-comparison)
+
+## Evaluation (trial) licenses
+
+You are [free to install](https://mui.com/store/legal/mui-x-eula/#evaluation-trial-licenses) and try our **Pro** / **Premium** components as long as it is not used for the development of a feature intended for production.
+Please take the component for a test run, no need to contact us.
+
+## License key
+
+For commercially licensed software, a license key is provided.
+This removes the watermark and console warning when valid.
+This license key is meant as a reminder for developers and their team to know when they forgot to license the software or renew the license.
+
+### License key installation
+
+Once you purchase a license, you'll receive a license key by email.
+This key should be installed to remove the watermark and
+console warnings.
+You must set the license key before rendering the first component.
+You only need to install the key once in your application.
+
+```jsx
+import { LicenseInfo } from '@mui/x-license-pro';
+
+LicenseInfo.setLicenseKey(
+  'x0jTPl0USVkVZV0SsMjM1kDNyADM5cjM2ETPZJVSQhVRsIDN0YTM6IVREJ1T0b9586ef25c9853decfa7709eee27a1e',
+);
+```
+
+### Security
+
+The check of the license key is done without making any network requests.
+
+The license key is designed to be public, the only thing we ask of licensed users is to not proactively publicize their license key.
+Exposing the license key in a JavaScript bundle is expected.
+
+### Validation errors
+
+If the validation of the license key fails, the component displays a watermark and provides a console warning in both development and production.
+End users can still use the component.
+
+Here are the different possible validation errors:
+
+#### Missing license key
+
+If the license key is missing, the component will look something like this:
+
+<div class="only-light-mode">
+  <img src="/static/x/watermark-light.png" style="width: 653px; margin-bottom: 2rem;" alt="" loading="lazy">
+</div>
+<div class="only-dark-mode">
+  <img src="/static/x/watermark-dark.png" style="width: 645px; margin-bottom: 2rem;" alt="" loading="lazy">
+</div>
+
+> Note that you are still allowed to use the component for [evaluation purposes](#evaluation-trial-licenses) in this case.
+
+#### License key expired
+
+The licenses are perpetual, the license key will work forever with the current version of the software.
+
+However, **access to updates/upgrades** is not perpetual.
+Installing a version of the component released after the license key has expired will trigger a watermark and console message.
```

Comment:
Rephrasing and expanding on this for clarity.

```suggestion
An expired license key will always work with the version of the component it was licensed to cover.
But if you try to install a newer version of the component with an expired license, it will display a watermark and a console warning.
```